posts/2022-12-20-22.0.0.13.adoc

```diff
@@ -0,0 +1,192 @@
+|21.0.0.2 - 22.0.0.12
+|Affects the link:{url-prefix}/docs/latest/reference/feature/grpc-1.0.html[grpc-1.0] and link:{url-prefix}/docs/latest/reference/feature/grpcClient-1.0.html[grpcClient-1.0] features
+|===
+
+For a list of past security vulnerability fixes, reference the link:{url-prefix}/docs/latest/security-vulnerabilities.html[Security vulnerability (CVE) list].
+
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A220013+label%3A%22release+bug%22[full list of bugs fixed in 22.0.0.13].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23478[NullPointerException in InstallFeatureAction for .esa files]
++
+A bug introduced in the featureUtility command caused a `NullPointerException` to occur when trying to install a feature with using the Enterprise Subsystem Archive (ESA) packaging denoted by an `.esa` extension.
++
+[source]
+----
+java.lang.NullPointerException
+        at com.ibm.ws.install.internal.InstallUtils.getFeatureName(InstallUtils.java:794)
+        at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.esaInstallInit(InstallFeatureAction.java:188)
+        ....
+----
++
+This is has been resolved, and invoking `./featureUtility installFeature x.esa` to install ESA features no longer throw an NPE.
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23403[HTTP/2 Intermittent server quiesce failure when stream is closed with an exception]
++
+An intermittent issue can occur causing threads to hang when an exception occurs on an HTTP/2 connection. This results in a quiesce warning being issued during a server stop.
++
+The issue has been resolved and the HTTP/2 streams and connections and the threads managing now stop without error, even in exception conditions.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23326[Liberty default HttpAuthenticationMechanisms do not call HttpMessageContext.responseUnauthorized]
++
+`HttpMessageContextWrapper` should be able to override the `responseUnauthorized` method. Due to a bug, Liberty's `HttpAuthenticationMechanisms` did not call `HttpMessageContext.responseUnauthorized` preventing users from to enriching the response, like adding headers.
++
+This issue has been resolved and the default `HttpAuthenticationMechanisms` now properly calls `HttpMessageContext.responseUnauthorized` for unauthorized requests.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23146[JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion() return incorrect version]
++
+If a Pages, formerly known as JSP,  application calls `JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion()` then a wrong version is returned. The value is coded to return `2.1` rather than the correct specification version.  
```

Comment:
```suggestion
If a Pages, formerly known as JSP,  application calls `JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion()`, the wrong version is returned. The value is coded to return `2.1` rather than the correct specification version.  
```